internal/driver/glfw/driver_desktop.go

```diff
@@ -136,10 +136,12 @@ func (d *gLDriver) refreshSystray(m *fyne.Menu) {
 	defer d.systrayLock.Unlock()
 
 	d.systrayMenu = m
-	systray.ResetMenu()
-	d.refreshSystrayMenu(m, nil)
+	runOnMain(func() {
+		systray.ResetMenu()
+		d.refreshSystrayMenu(m, nil)
```

Comment:
I think we have a double `runOnMain` here since `refreshSystrayMenu` also does a `runOnMain`? Also I believe the systray unlock has to be moved inside the runOnMain block here to prevent another call to refreshSystray from starting until the update is fully complete?

Reply:
Alternatively, get rid of the lock entirely, since we now have the runOnMain queue to enforce atomicity, but move d.systrayMenu = m inside as well?

Reply:
Nice suggestion, thanks